src/Compiler/Driver/GraphChecking/FileContentMapping.fs

```diff
@@ -9,10 +9,18 @@ type Continuations = ((FileContentEntry list -> FileContentEntry list) -> FileCo
 let collectFromOption (mapping: 'T -> 'U list) (t: 'T option) : 'U list = List.collect mapping (Option.toList t)
 
 let longIdentToPath (skipLast: bool) (longId: LongIdent) : LongIdentifier =
-    match skipLast, longId with
-    | true, _ :: _ -> List.take (longId.Length - 1) longId
-    | _ -> longId
-    |> List.map (fun ident -> ident.idText)
+
+    let rec loop skipLast (longId: LongIdent) : LongIdent =
+        match skipLast, longId with
+        | true, h :: t when h.idText = "`global`" ->
```

Comment:
why `true` and not `_` for the `skipLast` param in the global case?

Reply:
`false` is passed for module abbreviations, where you can use global like this as well:

```fsharp
module L = global.FSharp.Collections.List
```

Reply:
Yes, overlooked that, thanks

Reply:
> Is the recursion really worth it? I don't think you can have `global.global` or `Foo.global.Blah`, so maybe just shadow `longId` after the initial check.

I just didn't want to overcomplicate checks, since I can't really have a case with `when` and other case without in OR pattern, to check them in one branch. I also need to know if `global` is there, there are needed count of elements for `List.take` not to fail/not to return empty list, I suppose, I felt like recursive approach fits there well.

Reply:
Changed it to separate check